agent/taskresource/types_windows_mac.go

```diff
@@ -19,11 +19,13 @@ import (
 	"context"
 
 	"github.com/aws/amazon-ecs-agent/agent/dockerclient/dockerapi"
+	s3factory "github.com/aws/amazon-ecs-agent/agent/s3/factory"
 )
 
 // ResourceFields is the list of fields required for creation of task resources
 type ResourceFields struct {
 	*ResourceFieldsCommon
-	Ctx          context.Context
-	DockerClient dockerapi.DockerClient
+	Ctx             context.Context
+	DockerClient    dockerapi.DockerClient
+	S3ClientCreator s3factory.S3ClientCreator
```

Comment:
This is for windows_mac. The right move should be have a types_windows.go and types_unsupported.go and only add S3ClientCreator in windows.go.

Reply:
Done!